server/src/main/java/thesistrack/ls1/service/ThesisService.java

```diff
@@ -265,6 +266,40 @@ public Resource getThesisFile(Thesis thesis) {
         return uploadService.load(filename);
     }
 
+    public Thesis createPresentation(User creator, Thesis thesis, ThesisPresentationType type, ThesisPresentationVisibility visibility, String location, String streamUrl, Instant date) {
+        ThesisPresentation presentation = new ThesisPresentation();
+
+        presentation.setThesis(thesis);
+        presentation.setType(type);
+        presentation.setVisibility(visibility);
+        presentation.setLocation(location);
+        presentation.setStreamUrl(streamUrl);
+        presentation.setScheduledAt(date);
+        presentation.setCreatedBy(creator);
+        presentation.setCreatedAt(Instant.now());
```

Comment:
Those attributes are not validated here and could be tampered with on client side. In the `ThesisPresentation` class, there are constraints using the `@` notation - do those throw errors when violated? 

You could also get the currently authenticated user for the `createdBy` value instead of accepting the value provided as argument.

Reply:
I have refactored my validation a bit since the last PR:
- I do payload validation in the controller and the service just accepts anything to make it more flexible and usable for different controllers
- The authentication user is also only fetched in the controller. If I would want to get the authenticated user in the service, I would have to pass the JwtToken down to it which I dont want. I think its better if the controller does the authorization and access checks and the service just executes what it should